protocol/common/message_test.go

```diff
@@ -250,6 +250,339 @@ func TestConvertFromProtoToLinkPreviews(t *testing.T) {
+	require.NotNil(t, l2.GetCommunity())
+	require.Nil(t, l2.GetContact())
+	require.Nil(t, l2.GetChannel())
+	c2 := l2.GetCommunity()
+	require.Equal(t, community.CommunityID, c2.CommunityId)
+	require.Equal(t, community.DisplayName, c2.DisplayName)
+	require.Equal(t, community.Description, c2.Description)
+	require.Equal(t, community.MembersCount, c2.MembersCount)
+	require.Equal(t, community.Color, c2.Color)
+	require.Equal(t, community.TagIndices, c2.TagIndices)
+	require.NotNil(t, c2.Icon)
+	require.Equal(t, uint32(community.Icon.Width), c2.Icon.Width)
+	require.Equal(t, uint32(community.Icon.Height), c2.Icon.Height)
+	require.Equal(t, expectedThumbnailPayload, c2.Icon.Payload)
+	require.NotNil(t, c2.Banner)
+	require.Equal(t, uint32(community.Banner.Width), c2.Banner.Width)
+	require.Equal(t, uint32(community.Banner.Height), c2.Banner.Height)
+	require.Equal(t, expectedThumbnailPayload, c2.Banner.Payload)
+
+	// Channel link
+
+	l3 := unfurledLinks.UnfurledStatusLinks[2]
+	require.Equal(t, "https://status.app/cc/", l3.Url)
+	require.NotNil(t, l3.GetChannel())
+	require.Nil(t, l3.GetContact())
+	require.Nil(t, l3.GetCommunity())
+
+	c3 := l3.GetChannel()
+	require.Equal(t, channel.ChannelUUID, c3.ChannelUuid)
+	require.Equal(t, channel.Emoji, c3.Emoji)
+	require.Equal(t, channel.DisplayName, c3.DisplayName)
+	require.Equal(t, channel.Description, c3.Description)
+	require.Equal(t, channel.Color, c3.Color)
+
+	require.NotNil(t, c3.Community)
+	require.Equal(t, channel.Community.CommunityID, c3.Community.CommunityId)
+	require.Equal(t, channel.Community.DisplayName, c3.Community.DisplayName)
+	require.Equal(t, channel.Community.Color, c3.Community.Color)
+	require.Equal(t, channel.Community.Description, c3.Community.Description)
+	require.Equal(t, channel.Community.MembersCount, c3.Community.MembersCount)
+	require.NotNil(t, c3.Community.Icon)
+	require.Equal(t, uint32(channel.Community.Icon.Width), c3.Community.Icon.Width)
+	require.Equal(t, uint32(channel.Community.Icon.Height), c3.Community.Icon.Height)
+	require.Equal(t, expectedThumbnailPayload, c3.Community.Icon.Payload)
+	require.NotNil(t, c3.Community.Banner)
+	require.Equal(t, uint32(channel.Community.Banner.Width), c3.Community.Banner.Width)
+	require.Equal(t, uint32(channel.Community.Banner.Height), c3.Community.Banner.Height)
+	require.Equal(t, expectedThumbnailPayload, c3.Community.Banner.Payload)
+
+	// Test any invalid link preview causes an early return.
+	invalidContactPreview := contact
+	invalidContactPreview.PublicKey = ""
+	invalidPreview := message.StatusLinkPreviews[0]
+	invalidPreview.Contact = invalidContactPreview
+	message.StatusLinkPreviews = []StatusLinkPreview{invalidPreview}
+	_, err = message.ConvertStatusLinkPreviewsToProto()
+	require.ErrorContains(t, err, "invalid status link preview, url='https://status.app/u/'")
+}
+
+func TestConvertFromProtoToStatusLinkPreviews(t *testing.T) {
+
+	contact := &protobuf.UnfurledStatusContactLink{
+		PublicKey:   "1",
```

Comment:
I'd replace the numbers with readable names so that if the test fails it's clear what it's on.

Reply:
I was a bit lazy to create all the values and wanted to keep the diversity at the same time 😄 
I'll switch it to smth like `PublicKey_1`, sound good?